Editor/AGS.Editor/AGSEditor.cs

```diff
@@ -1400,6 +1400,7 @@ public bool SaveGameFiles()
             }
             Settings.RecentGames.Insert(0, recentGame);
             Settings.Save();
+            Factory.ComponentController.FindComponent<AGS.Editor.Components.FileCommandsComponent>()?.RefreshRecentGamesMenu();

```

Comment:
Right from the start, this is better done using event, where AGSEditor declares event, and FileCommandsComponent subscribes to event. This way AGSEditor does not need to know that it must call some component.

Reply:
That event gets called more than once when it's called, and it's named as being a limit. I can do that, but this function will be called quite a few times where this approach calls it only once on update.

Reply:
> That event gets called more than once when it's called, and it's named as being a limit. I can do that, but this function will be called quite a few times where this approach calls it only once on update.

I'm sorry, I don't understand. Which event is "called more than once", and what does "named as being a limit" means?
How it will be different between calling a function here and calling a event?

Reply:
I created a debouncer utility and used it to implement it on top of the `ListChangeUpdate` event and then subscribed to the event in `FileCommandsComponent`. There is an existing `ListChanged` for `RecentGames` but it's used for limiting it's size and every insertion or removal calls it (so it even calls itself once when it does a removal).

Reply:
Ah, I see now, you were referring to a "recentgames list changed" event inside the AppSettings.